### ISSUE TITLE
Use of uninitialized value in numeric, cannot install Proc::Daemon on CentOS 5.11 64bit

### ISSUE THREAD

**Issue by sparvu** [akreal/Proc-Daemon#5]
Im not able anymore to use and install Proc::Daemon on CentOS 5.11 64bit. This was working fine before:

0.20  Wed Jun 24 2015
    - GH#4: fix taint safe test (rouzier).
    - RT#103130: make tests less time-sensitive (Remi Collet, ppisar).

0.19  Sat Mar 21 2015
    - RT#102360: add file_umask option (Cyrille Mastchenko).

See here:

```
Running make for A/AK/AKREAL/Proc-Daemon-0.20.tar.gz
cp lib/Proc/Daemon.pm blib/lib/Proc/Daemon.pm
cp lib/Proc/Daemon.pod blib/lib/Proc/Daemon.pod
  AKREAL/Proc-Daemon-0.20.tar.gz
  make -- OK
Running make test
PERL_DL_NONLAZY=1 /opt/kronometrix/perl/bin/perl "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01_loadmodule.t .. ok
# Proc::ProcessTable is installed and did find the right PID for 'perl /var/tmp/.cpan/build/Proc-Daemon-0.20-qeuMS6/t/kid.pl': 11778
Use of uninitialized value in numeric eq (==) at t/02_testmodule.t line 129.

#   Failed test 'the 'umask.file' has right permissions'
#   at t/02_testmodule.t line 129.
# Looks like you failed 1 test of 19.
t/02_testmodule.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/19 subtests 
# Proc::ProcessTable is installed and did find the right PID for 'perl /var/tmp/.cpan/build/Proc-Daemon-0.20-qeuMS6/t/kid.pl': 11791
t/03_taintmode.t ... ok
```

Possible to have a fix for these issues !? I cant build anymore Kronometrix data recording module on Linux, FreeBSD.

Many thanks.

**Comment by sparvu** [comment]
I can confirm 0.19 works fine. Problems are coming from latest 0.20

**Comment by akreal** [comment]
Could you provide some details on your setup? I tried fresh CentOS 5.11 installation in VirtualBox and got OKs for all tests.
I see also bunch of OpenBSD failures on CPAN Testers and also can not reproduce it.

**Comment by sparvu** [comment]
Sorry for delay. This is a virtual machine running CentOS 5.11 64bit, on top of Xen. Problem seems 0.20 is very unstable on tests: sometimes is giving errors, sometimes not. I did downgrade to 0.19 and this release is better: I always get same results.

Something is broken after 0.19 -

**Comment by akreal** [comment]
I've added more checks and this may resolve the problem:
https://github.com/akreal/Proc-Daemon/tree/test-wait-for-file
If you can, please try it.
I'm going to release it when there is something else to release, or if I have confirmation that it helps.

**Comment by akreal** [comment]
This change is released in 0.21 and I hope it will work for you. If not, please reopen the issue.
